pkg/devices/host.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package devices
+
+import (
+	"github.com/jaypipes/ghw"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils"
+)
+
+type hostDevice struct {
+	types.APIDevice
+	vendorID   string
+	deviceCode string
+	driver     string
+}
+
+// NewHostDevice returns an instance of HostDevice interface
+// A list of DeviceInfoProviders can be set externally.
+// If empty, the default driver-based selection provided by ResourceFactory will be used
+func NewHostDevice(dev *ghw.PCIDevice, deviceID string, rFactory types.ResourceFactory,
+	rc *types.ResourceConfig, infoProviders []types.DeviceInfoProvider) (types.HostDevice, error) {
+	// Get driver info
+	driverName, err := utils.GetDriverName(dev.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use the default Information Provided if not
+	if len(infoProviders) == 0 {
+		infoProviders = append(infoProviders, rFactory.GetDefaultInfoProvider(deviceID, driverName))
+	}
+
+	apiDevice := NewAPIDevice(deviceID, infoProviders)
+
+	if !rc.ExcludeTopology {
```

Comment:
can this be handled in `NewAPIDevice` ?

Reply:
that is just get nodeNum here and let APIDevice handle the creation of Topology WDYT?